pkg/admission/validator/secret.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"context"
+	"fmt"
+
+	gcpvalidation "github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp/validation"
+	"github.com/gardener/gardener-extension-provider-gcp/pkg/gcp"
+
+	extensionswebhook "github.com/gardener/gardener/extensions/pkg/webhook"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type secret struct {
+	client client.Client
+}
+
+// NewSecretValidator returns a new instance of a secret validator.
+func NewSecretValidator() extensionswebhook.Validator {
+	return &secret{}
+}
+
+// InjectClient injects the given client into the validator.
+func (s *secret) InjectClient(client client.Client) error {
+	s.client = client
+	return nil
+}
+
+// Validate validates the given secret objects.
+func (s *secret) Validate(ctx context.Context, new, _ runtime.Object) error {
```

Comment:
Don't you want to compare the old and new secret and skip any further logic in case both are equal?